docs/dbt/connections.rst

```diff
@@ -0,0 +1,289 @@
+
+   .. tab:: Snowflake
+
+        .. note::
+            Cosmos currently only supports User/Password authentication to Snowflake. Additional authentication methods will be
+            added soon. If there is an authentication method that your organization requires, open a PR/Issue on the Cosmos GitHub
+            repo `here <https://github.com/astronomer/astronomer-cosmos>`_.
+
+        Here are the specific steps that you should take if you are configuring Cosmos to run against Snowflake:
+
+        1. Add Cosmos and Snowflake Provider to your `requirements.txt`
+
+            .. code-block:: text
+
+                astronomer-cosmos[dbt-snowflake]
+                apache-airflow-providers-snowflake
+
+        2. Create an Airflow Connection for Snowflake. Fore more information on creating the Snowflake Airflow Connection, see
+        Airflow's official documentation `here <https://airflow.apache.org/docs/apache-airflow-providers-snowflake/stable/connections/snowflake.html>`_.
+        In the Airflow UI, navigate to `Admin >> Connections` and create a new connection like this:
+
+            .. figure:: https://github.com/astronomer/astronomer-cosmos/raw/main/docs/_static/connections_doc/snowflake_airflow_connection.png
+               :width: 800
+
+               Creating a snowflake connection in Airflow
+
+        (Optional) If your team prefers storing connections as environment variables or in a supported secrets backend, the connection URI
+        for this example would look like this:
+
+            .. note::
+                The following variable value would need to be encoded. You can use a service like `urlencoder.org <https://www.urlencoder.org/>`_  to do that.
+
+            .. code-block:: text
+
+                AIRFLOW_CONN_SNOWFLAKE_DEFAULT=snowflake://user123:p@ssword!@/my_db_schema?__extra__={"account":+"gp54783",+"warehouse":+"my_warehouse",+"database":+"my_db",+"region":+"us-east-1",+"role":+"user123",+"insecure_mode":+false}
+
+        3. With the dependencies and connection in place, you can pass this connection to a DbtTaskGroup, DbtDag, or Dbt Operator
+        Class via the `conn_id` parameter:
+
+            .. code-block:: python
+
+                jaffle_shop = DbtTaskGroup(
+                    ...
+                    conn_id="snowflake_default",
+                )
+
+        .. note::
+            If you left the `Schema` field blank when configuring your Snowflake connection or if you would like to override it,
+            then specify the schema parameter via the `dbt_args`:
+
+            .. code-block:: python
+
+                jaffle_shop = DbtTaskGroup(
+                    conn_id="snowflake_default",
+                    dbt_args={
+                        "schema": "my_schema",
+                    }
+                )
+
+
+   .. tab:: Redshift
+
+        .. note::
+            Cosmos currently only supports User/Password authentication to Redshift. Additional authentication methods will be
```

Comment:
@pohek321 how hard would it be to add other auth systems to cosmos? Do we have issues created for this?

Reply:
@dimberman we'll have to identify the Airflow connection fields to determine if the user is doing IAM vs. user/password, then we'll have to map the Airflow connection fields to the appropriate profiles.yml fields. Opened an issue for different auth methods already:
- [Add authentication method for Redshift (IAM Authentication)](https://github.com/astronomer/astronomer-cosmos/issues/44)
- [Add authentication method for Snowflake (Key Pair Authentication)](https://github.com/astronomer/astronomer-cosmos/issues/42)
- [Add authentication method for BigQuery (Service Account File)](https://github.com/astronomer/astronomer-cosmos/issues/43)

Reply:
Marking this as resolved since we have issues tracking various auth methods that we plan on adding.